### PR TITLE
fix(storage-browser): add input element for adding files and one for adding a folder

### PR DESCRIPTION
**Pull request by timngyn**
#### Description of changes
- For now, added two `input` elements for selecting files and one for folder

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx [hunk 1]
```diff
@@ -212,27 +196,39 @@ export const UploadControls = (): JSX.Element => {
...
+  const initialRun = React.useRef(false);
 
   let tableData = tasks.map((task) => {
-    const folder = task.data.webkitRelativePath.lastIndexOf('/') + 1;
+    const { webkitRelativePath } = task.data;
+
+    const folder =
+      webkitRelativePath?.length > 0
+        ? webkitRelativePath.slice(0, webkitRelativePath.lastIndexOf('/') + 1)
+        : '/';
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:212]: updated this cause I saw tests were throwing errors when trying to access `lastIndexOf` the `webkitRelativePath` property when it was undefined. Realized the original logic for this was wrong and so I updated it here

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx [hunk 2]
```diff
@@ -316,7 +312,27 @@ export const UploadControls = (): JSX.Element => {
...
+        data-testid="amplify-folder-select"
+        type="file"
+        style={{ display: 'none' }}
+        onChange={({ target }) => {
+          handleFileSelect([...(target.files ?? [])]);
+        }}
+        // @ts-expect-error webkitdirectory is not typed
+        webkitdirectory=""
+        ref={handleFolderInput}
+      />
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:335]: Can we put these directly in the "add" controls so we can remove the need for the effect/additional handling logic?

Comment (timngyn) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:335]: just to make sure I understand the ask, are you saying I make a new component for each input with their respective buttons?

**Review by calebpollman — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:335]: Can be addressed later

**Review by hbuchel — APPROVED** [review]

**Pull request merged**